src/arrayMethodJoin.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let separatorValue = separator;
+
+    if (separatorValue === undefined) {
+      separatorValue = ',';
+    }
+
+    const arr = this;
+    let result = '';
+
+    if (arr.length === 0) {
+      return '';
+    }
+
+    for (const element of arr) {
+      const str = element === null
+      || element === undefined ? '' : element.toString();
```

Comment:
`element.toString();` -> regarding this part, I personally am not sure whether all data types have `.toString` method, it would be safer to user something like `String(element)` or  interpolate `${element}` 
```js
result += this[i] === null || this[i] === undefined ? '' : String(this[i])
```